src/OpenTelemetry.AutoInstrumentation/Logger/LogBuilderExtensions.cs

```diff
@@ -31,8 +31,9 @@ public static ILoggingBuilder AddOpenTelemetryLogs(this ILoggingBuilder builder)
     {
         try
         {
-            if (!(builder.Services is ServiceCollection services))
+            if (!(builder.Services is IServiceCollection services))
```

Comment:
In general, you can change the condition to builder.Services == null.

ILoggingBuilder Services is always `IServiceCollection`.

Shouldn't it check some extended supported set of types?

Reply:
Good thought, we could check this for null. I don't recall the reason for checking ServicesCollection here.

Reply:
Fixed.